dts/bindings/tcpc/st,stm32-ucpd.yaml

```diff
@@ -0,0 +1,78 @@
+# Copyright 2021 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+description: ST STM32 family USB Type-C / Power Delivery.
+             The default values were taken from the
+             LL_UCPD_StructInit function defined in
+             stm32cube/stm32u5xx/drivers/src/stm32u5xx_ll_ucpd.c,
+             stm32cube/stm32g0xx/drivers/src/stm32g0xx_ll_ucpd.c,
+             stm32cube/stm32l5xx/drivers/src/stm32l5xx_ll_ucpd.c, and
+             stm32cube/stm32g4xx/drivers/src/stm32g4xx_ll_ucpd.c.
+
+compatible: "st,stm32-ucpd"
+
+include: base.yaml
+
+properties:
+    label:
+      required: true
+
+    reg:
+      required: true
+
+    clocks:
+      required: true
+
+    interrupts:
+      required: false
+      description:
+        When false, the subsystem using a compatible driver is poll driven
```

Comment:
> When false [....]

Blocking comment:
Interrupt property can't be false, but can be "not provided (/not available)".

Non blocking comment:
IIUC, you mean that, interrupt property is not mandatory since driver is able to detect missing interrupt property and behave as poll driven in that case, correct?
But, then do you expect user to explicitly remove interrupt property if he/she wants explicitly to use driver in polling mode ? Or is it rather a driver flexibilty but user is not supposed to mess with that.

If that's a choice which is left to the user, I'd suggest to use a Kconfig for this kind of driver behavior configuration (which doesn't prevent to have some kind of DT magic to control dts description is matching expected behavior (ie irq available for irq driven mode).

Reply:
I agree that the option of polling vs interrupt should be placed in a Kconfig and I'll change the interrupt property to "required: true".